core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -837,51 +836,54 @@ public void run() {
             nextBackoffIntervalNanos = retryPolicy.initialBackoffNanos;
           }
 
-          RetryPlan retryPlan = makeRetryDecision(status, trailers);
-          if (retryPlan.shouldRetry) {
-            // The check state.winningSubstream == null, checking if is not already committed, is
-            // racy, but is still safe b/c the retry will also handle committed/cancellation
-            FutureCanceller scheduledRetryCopy;
+          if (!isHedging) {
```

Comment:
Nit: Swap the blocks here to avoid the negation, so that this can be `if (isHedging) { /* handle hedging */ } else { /* handle retry */ }`?

Reply:
Done.